### ISSUE TITLE
Availability recovery seems to be having issues

### ISSUE THREAD

**Issue by eskimor** [paritytech/polkadot#5246]
As we can see [here](https://grafana.parity-mgmt.parity.io/goto/bfi6C9snk?orgId=1)

![Screenshot from 2022-04-04 13-35-29](https://user-images.githubusercontent.com/1527017/161535623-ba038409-e161-4d09-8f13-5b4909d39cea.png)

we are issuing 4 times the amount of request that get actually finished on Versi. 

On Kusama it is still 3 times: https://grafana.parity-mgmt.parity.io/goto/Z9kWq9snz?orgId=1
![Screenshot from 2022-04-04 13-38-30](https://user-images.githubusercontent.com/1527017/161536039-abc18731-3d39-4753-bf5f-05f0a7af9b2c.png)

 Under normal conditions a few percent more requests than finished requests can be quite likely be explained, but not a factor of 3-4. Something weird is going on here.

**Comment by vstakhov** [comment]
These requests are measured with this timer in the future: https://github.com/paritytech/polkadot/blob/master/node/network/availability-recovery/src/lib.rs#L334

However, once we recover all chunks that are required to restore the availability data, the remaining futures are just dropped without any other metrics being touched: https://github.com/paritytech/polkadot/blob/master/node/network/availability-recovery/src/lib.rs#L449

So this rate actually depends on the validator count: we need 1/3 of chunks to recover the availability data, so the threshold is calculated as following: https://github.com/paritytech/polkadot/blob/master/erasure-coding/src/lib.rs#L87 and then used to calculate how many requests we should send: https://github.com/paritytech/polkadot/blob/master/node/network/availability-recovery/src/lib.rs#L280

So basically for higher keys rotation count (and temporary errors that can take place on such an event) we would send more requests in general.

Hence, upon this investigation I'm under impression that it works as expected.

**Comment by eskimor** [comment]
Thanks @vstakhov! Unfortunately, I cannot really follow your argument: First: keys are almost never rotated in practice, second: certainly sending out more requests than strictly necessary is absolutely expected, but not by a factor of 3 to 4, that still looks fishy to me.

**Comment by vstakhov** [comment]
According to our recent conversation, it seems that the usage of `avg` was not a good choice as `chunk_requests_finished` is actually a vector with different labels. Changing that to `sum` seems to show the actual rate that is roughly equal to 1. Spot checking of the subsystem logic has not revealed any misbehaviour as well. Therefore, I think this issue might be closed.

**Comment by eskimor** [comment]
Thanks @vstakhov ! Awesome!
